src/lists/list.jsx

```diff
@@ -92,35 +95,32 @@ const List = React.createClass({
       ...other,
     } = this.props;
 
+    let hasSubheader = false;
+
+    if (subheader) {
+      hasSubheader = true;
+    } else {
+      const firstChild = React.Children.toArray(children)[0];
+      if (React.isValidElement(firstChild) && firstChild.type === Subheader) {
```

Comment:
Good to know...will watch out for areas we're doing this. This approach is a lot better/safer IMO.